content/operate/guides/cosmos/syncing.md

```diff
@@ -4,6 +4,242 @@ title: Syncing
+2022-05-30T22:39:38.819+0100 (firehose.hub.js.file) reading from blocks store: file does not (yet?) exist, retrying in (bstream/filesource.go:194) {"filename": "/../fh-data/storage/merged-blocks/0005200700.dbin.zst", "base_filename": "0005200700", "retry_delay": "4s", "secondary_blocks_stores_count": 0}
+10:39PM INF exiting...
+2022-05-30T22:39:39.414+0100 (ingestor) mindreader is stopping (mindreader/mindreader.go:241)
+2022-05-30T22:39:39.414+0100 (ingestor) waiting until consume read flow (i.e. blocks) is actually done processing blocks... (mindreader/mindreader.go:255)
+2022-05-30T22:39:39.414+0100 (ingestor) reached end of console reader stream, nothing more to do (mindreader/mindreader.go:228)
+2022-05-30T22:39:39.414+0100 (ingestor) all blocks in channel were drained, exiting read flow (mindreader/mindreader.go:269)
+2022-05-30T22:39:39.414+0100 (ingestor) archiver selector is terminating (mindreader/archiver_selector.go:88)
+2022-05-30T22:39:39.414+0100 (ingestor) merger archiver is terminating (mindreader/merge_archiver.go:65)
+2022-05-30T22:39:39.414+0100 (ingestor) terminating upload loop (mindreader/merge_archiver.go:100)
+2022-05-30T22:39:39.414+0100 (ingestor) merger archiver is terminated (mindreader/merge_archiver.go:73)
+2022-05-30T22:39:39.414+0100 (ingestor) one block archiver is terminating (mindreader/archiver.go:66)
+2022-05-30T22:39:39.414+0100 (ingestor) terminating upload loop (mindreader/archiver.go:146)
+2022-05-30T22:39:39.414+0100 (ingestor) one block archiver is terminated (mindreader/archiver.go:74)
+2022-05-30T22:39:39.414+0100 (ingestor) archiver selector is terminated (mindreader/archiver_selector.go:98)
+2022-05-30T22:39:39.414+0100 (ingestor) archiver Terminate done (mindreader/mindreader.go:275)
+2022-05-30T22:39:39.414+0100 (ingestor) consume read flow terminate (mindreader/mindreader.go:257)
+2022-05-30T22:39:39.414+0100 (dfuse) All apps terminated gracefully (launcher/launcher.go:273)
+```
+
+---
+
+## Overview and Explanation
+
+The `ingestor` process consumes data from the node and saves it as one-blocks files.
+
+The `merger` process downloads these single block files and creates bigger 100 blocks files inside the `merged-block` folder with a naming convention that matches the blocks inside the file. e.g. `0005200700.dbin.zst` contains the blocks from `5200700 - 5200799`.
+
+The `relayer` process uses gRPC to connect to the `ingestor` process and consume real-time block updates. It can also use the `merger` as a back-up mechanism in the case there are missing files.
+
+The `firehose` process will connect to the `relayer(s)` via gRPC and to the data store in order to provide a live feed of blocks in a joined manner. The end consumer can connect to it via gRPC to read from the stream. It uses the merged-block files created by the `merger` process to serve this data to the services downstream.
```

Comment:
This part is the same for all Firehoses, we could link to a page where it is explained (or skip it entirely).